x-pack/platform/plugins/shared/onechat/server/services/chat/chat_service.ts

```diff
@@ -124,83 +125,97 @@ class ChatServiceImpl implements ChatService {
-          conversationId,
-          conversationClient,
-        });
-        const agentEvents$ = getExecutionEvents$({
-          agentId,
-          request,
-          mode,
-          conversation$,
-          nextInput,
-          agentService: this.agentService,
-        });
 
-        const title$ = isNewConversation
-          ? generatedTitle$({ chatModel, conversation$, nextInput })
-          : conversation$.pipe(
-              switchMap((conversation) => {
-                return of(conversation.title);
-              })
-            );
+          const title$ = isNewConversation
+            ? generatedTitle$({ chatModel, conversation$, nextInput })
+            : conversation$.pipe(
+                switchMap((conversation) => {
+                  return of(conversation.title);
+                })
+              );
 
-        const roundCompletedEvents$ = agentEvents$.pipe(filter(isRoundCompleteEvent));
+          const roundCompletedEvents$ = agentEvents$.pipe(filter(isRoundCompleteEvent));
 
-        const saveOrUpdateAndEmit$ = isNewConversation
-          ? createConversation$({
-              agentId,
-              conversationClient,
-              title$,
-              roundCompletedEvents$,
-            })
-          : updateConversation$({
-              conversationClient,
-              conversation$,
-              title$,
-              roundCompletedEvents$,
-            });
+          const saveOrUpdateAndEmit$ = isNewConversation
+            ? createConversation$({
+                agentId,
+                conversationClient,
+                title$,
+                roundCompletedEvents$,
+              })
+            : updateConversation$({
+                conversationClient,
+                conversation$,
+                title$,
+                roundCompletedEvents$,
+              });
 
-        return merge(agentEvents$, saveOrUpdateAndEmit$).pipe(
-          catchError((err) => {
-            this.logger.error(`Error executing agent: ${err.stack}`);
-            return throwError(() =>
-              isOnechatError(err)
-                ? err
-                : createInternalError(`Error executing agent: ${err.message}`, {
+          return merge(agentEvents$, saveOrUpdateAndEmit$).pipe(
+            catchError((err) => {
+              this.logger.error(`Error executing agent: ${err.stack}`);
+              return throwError(() => {
+                const traceId = getCurrentTraceId();
+                if (isOnechatError(err)) {
+                  err.meta = {
+                    ...err.meta,
+                    traceId,
+                  };
+                  return err;
+                } else {
+                  return createInternalError(`Error executing agent: ${err.message}`, {
                     statusCode: 500,
-                  })
-            );
-          }),
-          shareReplay()
-        );
-      })
-    );
+                    traceId,
+                  });
```

Comment:
attaching the traceId to the error to be able to surface it from the HTTP API call.